syntax = "proto3";

import "google/protobuf/empty.proto";

message FileComponent {
    int32 id = 1;
    string repository_id = 2;
    string file_path = 3;
    int32 start_line = 4;
    int32 end_line = 5;
    string content = 6;
}

message FileComponentIds {
    repeated int32 file_component_ids = 1;
}

message RepositoryFilePaths {
    string repository_id = 1;
    repeated string file_paths = 2;
}

message DeleteFileComponentsByRepositoryIdRequest {
    string repository_id = 1;
}

message DeleteFileComponentsByRepositoryIdAndFilePathsRequest {
    string repository_id = 1;
    repeated string file_paths = 2;
}

service FileComponentsService {
    rpc CreateFileComponents(RepositoryFilePaths) returns(stream FileComponent);
    rpc GetFileComponents(FileComponentIds) returns (stream FileComponent);
    rpc DeleteFileComponentsByRepositoryId(DeleteFileComponentsByRepositoryIdRequest) returns (google.protobuf.Empty);
    rpc DeleteFileComponentsByRepositoryIdAndFilePaths(DeleteFileComponentsByRepositoryIdAndFilePathsRequest) returns (FileComponentIds);
}